### PR TITLE
chore: use pngquant to optimize PNG images

### PR DESCRIPTION
**Pull request by dsanders11**
Uses [`imagemin-pngquant`](https://github.com/imagemin/imagemin-pngquant) to optimize PNG images at build time.

Significantly reduces the size of images used on the home page, shrinking transferred size from 4.4 MB to 1.2 MB.

It's technically lossy optimization, but I don't think anyone can tell the difference between [this](https://www.electronjs.org/assets/marketing/app-store.png) and [this](https://electronjsorg-new-pr-583.herokuapp.com/assets/marketing/app-store.png) unless you've got Superman eyes.

**Before**:
Before

**After**:
After

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,9 @@
     "docusaurus-plugin-sass": "^0.2.1",
     "dotenv": "^16.0.3",
     "execa": "^5.0.0",
+    "image-minimizer-webpack-plugin": "^4.0.2",
+    "imagemin": "^9.0.0",
+    "imagemin-pngquant": "^10.0.0",
```
Comment (BlackHole1) [inline package.json:45]: It seems it should be moved to `devDependencies`.

**Review by BlackHole1 — DISMISSED** [review]

**Comment by dsanders11** [comment]
The compelling reason is the significant reduction in size for imperceptible visual change.

I briefly tied a lossless optimizer on one of the larger files as a test and it was something like a 15% reduction, compared to 75% with pngquant.

**Review by erickzhao — APPROVED** [review]
Great idea. LGTM once @BlackHole1's comment is addressed. :)

**Comment by dsanders11** [comment]
Had an offline discussion with @ckerr regarding his comment, and moving ahead with taking the easy win here and we can follow up with other improvements in the future, such as committing losslessly optimized versions of the files to begin with. 👍

**Pull request merged**